### DISCUSSION TITLE
Slow initialization and stuck in initialization phase when using OSQP for quadratic programming

### DISCUSSION THREAD

**Discussion by ineffable-cue** [osqp/osqp#696] [Python Interface] [answered]
I am using OSQP to solve a quadratic programming problem. However, I am experiencing extremely slow performance and the process seems to be stuck in the initialization phase.
The size of matrix A is (3410326, 79921) and the size of matrix P is (79921, 79921).
I am looking for suggestions on how to improve the performance or resolve this issue. Any help would be greatly appreciated.
![image](https://github.com/user-attachments/assets/ce089a54-beaf-48df-8de3-5c52470606da)

**Comment by goulart-paul** [comment]
How many nonzeros are in A and P?

**Reply by ineffable-cue** [reply]
Thank you for your reply. The number of nonzeros in A is 6,988,620, and the number of nonzeros in P is 25,394,457.

**Reply by goulart-paul** [reply, marked as answer]
The number of nonzeros is really large in your problem, and its likely that the issue is that the internal memory allocation and factorisation of the KKT conditions for your problem are consequently very slow.    This is partly (maybe entirely) because the internal sparse matrix factorisation method defaults to the QDLDL package, which is a simplicial matrix factorisation method that doesn't always scale well to very large problems.

If you are on an intel platform then you could try using the MKL solver backend instead, or even the indirect / conjugate gradient based linear solver method.   I'm not sure if those are in the current release (0.6) though, or only in the 1.0 beta release.

**Reply by ineffable-cue** [reply]
Thank you. I tried the suggestions you gave and successfully solved the problem.
